### ISSUE TITLE
vigranumpy tests segfaulting in master

### ISSUE THREAD

**Issue by psteinb** [ukoethe/vigra#346]
Hi, I am new to this library, so please excuse me if I am overseeing something obvious. I wanted to build vigra master and ran to a curious problem.
```
$ cmake  ..
$ make
$ make check
...
Running vigranumpy tests

executing test file /lustre/projects/hpcsupport/steinbac/software/vigra/master-src/build/vigranumpy/test/test1.pyc
E
executing test file /lustre/projects/hpcsupport/steinbac/software/vigra/master-src/build/vigranumpy/test/test2.pyc
./run_vigranumpytest.sh: line 2:  1912 Segmentation fault      (core dumped) /sw/apps/python/2.7.9/bin/python -c "import nose; nose.main()" .
make[3]: *** [vigranumpy/test/vigranumpytest.so] Error 1
make[3]: *** Deleting file `vigranumpy/test/vigranumpytest.so'
make[2]: *** [vigranumpy/test/CMakeFiles/vigranumpytest.dir/all] Error 2
make[1]: *** [CMakeFiles/check.dir/rule] Error 2
make: *** [check] Error 2
```
I then build vigra with debug symbols through and called the nose test runner through gdb:
```
$ cmake -DCMAKE_BUILD_TYPE=Debug ..
$ make 
$ make check
$ cd vigranumpy/test
$ gdb --args python -c "import nose; nose.main()"
```
The stack trace on the segfault about points to:
```
(gdb) bt
#0  0x00000000000f2c06 in ?? ()
#1  0x00002aaab663b717 in initvigranumpycore () at /lustre/projects/hpcsupport/steinbac/software/vigra/master-src/vigranumpy/src/core/vigranumpycore.cxx:67
#2  0x00002aaaaabd11f9 in _PyImport_LoadDynamicModule (name=0x996060 "vigra.vigranumpycore", pathname=0xc3c400 "/lustre/projects/hpcsupport/steinbac/software/vigra/master-src/build/vigranumpy/vigra/vigranumpycore.so", fp=<optimized out>) at ./Python/importdl.c:53
#3  0x00002aaaaabceda9 in import_submodule (mod=mod@entry=0x2aaab6333b40, subname=subname@entry=0x996066 "vigranumpycore", fullname=fullname@entry=0x996060 "vigra.vigranumpycore") at Python/import.c:2700
...
```
Any ideas?

PS. CentOS release 6.3, gcc/4.9.2, python/2.7.9, boost/1.59.0 and fftw/3.3.4

**Comment by psteinb** [comment]
just to add: my personal solution is to build without numpy support even though that does not appear to be ideal.

**Comment by ukoethe** [comment]
Hi Peter,

on first glance this pretty much looks like a problem with the PATH. 
That is, you compiled against one version of a dependency, but at 
runtime a shared library of another version is found in the PATH, 
leading to an immediate crash at startup.

Best
Ulli

Am 08.03.2016 um 11:08 schrieb Peter Steinbach:

**Comment by psteinb** [comment]
Yeah I suspected this, once I had a look at your test scripts. This is a 
cluster installation of vigra, I am working on. We have the system 
packages installed in the standard paths and anything more recent is 
made available through the environment (PATH, LD_LIBRARY_PATH, 
PYTHON_PATH, ...). And believe me there is a lot of packages that 
shipped recent versions than in the CentOS 6.3 repos.

CTest allows you to specify environment variables for the shell that 
runs the tests. Can your test scripts be upgraded to a similar behavior?

**Comment by ukoethe** [comment]
It should be sufficient to set these variables in the shell before 
calling 'make check'. Just make sure that everything is consistent with 
the cmake configuration as shown by 'cmake-gui'. You can do 'ldd' on the 
python modules (especially vigranumpytest.pyd) to confirm that the 
expected shared libraries are being used.

Best
Ulli

Am 08.03.2016 um 11:54 schrieb Peter Steinbach:
